### PR TITLE
Recover for cases that need to be truncate

### PR DESCRIPTION
**Pull request by ever0de**

### CODE REVIEW

File: src/xxh3.rs [hunk 1]
```diff
@@ -20,6 +20,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
+#![allow(clippy::cast_possible_truncation)]
```
Comment (cberner) [inline src/xxh3.rs:23]: Instead of allowing it for the whole file, can you move these to just the two functions that need it?

**Review by cberner — CHANGES_REQUESTED** [review]

File: src/xxh3.rs (hunk 1 above)
Comment (ever0de) [reply src/xxh3.rs:23]: Sure

**Review by cberner — APPROVED** [review]

**Comment by cberner** [comment]
Hmm, I think there are a lot of cases that are still broken. Can you try running the fuzzer via `make fuzz`?

**Comment by cberner** [comment]
I'm going to revert these commits, since they broke master. I saved a copy of the master branch as the branch `master-backup`. Can you resubmit them as a new PR?

**Comment by cberner** [comment]
I also have a PR open to run the fuzzer as part of CI, so hopefully that will make these changes easier the second time around

**Comment by ever0de** [comment]
The `make fuzz` command did not work, so I shoveled. I'll check and fix it and open a PR.
ref: #7

**Pull request merged**